toolchain/check/testdata/namespace/no_prelude/name_poisoning.carbon

```diff
@@ -0,0 +1,386 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/namespace/no_prelude/name_poisoning.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/namespace/no_prelude/name_poisoning.carbon
+
+// --- no_poison.carbon
+
+library "[[@TEST_NAME]]";
+
+namespace N1;
+
+// N2.N3 uses N2.N1 and not N1.
```

Comment:
Perhaps put two functions with the same name but different return types into those two namespaces, and then call `N2.N3.F()` to make sure you get the expected one.

Reply:
I've created a class and made sure it can be used to refer to the same class in both of the correct namespaces.